spec/devise_saml_authenticatable/strategy_spec.rb

```diff
@@ -35,15 +31,55 @@
     end
 
     it "authenticates with the response" do
-      expect(OneLogin::RubySaml::Response).to receive(:new).with(params[:SAMLResponse], settings: saml_config)
+      expect(OneLogin::RubySaml::Response).to receive(:new).with(params[:SAMLResponse], anything)
       expect(user_class).to receive(:authenticate_with_saml).with(response)
       expect(user).to receive(:after_saml_authentication).with(response.sessionindex)
 
       expect(strategy).to receive(:success!).with(user)
       strategy.authenticate!
     end
 
-    context "and the resource cannot does not exist" do
+    context "when saml config uses an idp_adapter" do
+      let(:idp_providers_adapter) {
+        Class.new {
+          def self.settings(idp_entity_id)
+            #something about this spec's setup is keeping the code in here from being used
+            {
+              assertion_consumer_service_url: "acs_url",
+              assertion_consumer_service_binding: "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+              name_identifier_format: "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
+              issuer: "sp_issuer",
+              idp_entity_id: "http://www.example.com",
+              authn_context: "",
+              idp_slo_target_url: "idp_slo_url",
+              idp_sso_target_url: "http://idp_sso_url",
+              idp_cert: "idp_cert"
+            }
+          end
+        }
+      }
+
+      before do
+        Devise.idp_settings_adapter = idp_providers_adapter
+        allow(idp_providers_adapter).to receive(:settings).and_return({})
+      end
+
+      it "is valid" do
+        expect(strategy).to be_valid
+      end
+
+      it "authenticates with the response for the corresponding idp" do
```

Comment:
The authenticate method new authenticates for the corresponding IdP after the adapter looks up the settings by entity id. This method is called in the controller `create` action. The create action is not tested in the `SamlSessionsController` specs, however, because it inherits all of its behavior from `Devise::SessionsController`